src/request/providers/node.ts

```diff
@@ -0,0 +1,558 @@
+
+	return true;
+}
+
+export default function node(url: string, options: NodeRequestOptions = {}): Task<Response> {
+	const parsedUrl = urlUtil.parse(options.proxy || url);
+
+	const requestOptions: HttpsOptions = {
+		agent: options.agent,
+		auth: parsedUrl.auth || options.auth,
+		ca: options.ca,
+		cert: options.cert,
+		ciphers: options.ciphers,
+		host: parsedUrl.host,
+		hostname: parsedUrl.hostname,
+		key: options.key,
+		localAddress: options.localAddress,
+		method: options.method ? options.method.toUpperCase() : 'GET',
+		passphrase: options.passphrase,
+		path: parsedUrl.path,
+		pfx: options.pfx,
+		port: Number(parsedUrl.port),
+		rejectUnauthorized: options.rejectUnauthorized,
+		secureProtocol: options.secureProtocol,
+		socketPath: options.socketPath
+	};
+
+	requestOptions.headers = <{ [key: string]: string }> options.headers || {};
+
+	if (!Object.keys(requestOptions.headers).map(headerName => headerName.toLowerCase()).some(headerName => headerName === 'user-agent')) {
+		requestOptions.headers[ 'user-agent' ] = 'dojo/' + version + ' Node.js/' + process.version.replace(/^v/, '');
+	}
+
+	if (options.proxy) {
+		requestOptions.path = url;
+		if (parsedUrl.auth) {
+			requestOptions.headers[ 'proxy-authorization' ] = 'Basic ' + new Buffer(parsedUrl.auth).toString('base64');
+		}
+
+		const parsedProxyUrl = urlUtil.parse(url);
+		if (parsedProxyUrl.host) {
+			requestOptions.headers[ 'host' ] = parsedProxyUrl.host;
+		}
+		requestOptions.auth = parsedProxyUrl.auth || options.auth;
+	}
+
+	if (!options.auth && (options.user || options.password)) {
+		requestOptions.auth = encodeURIComponent(options.user || '') + ':' + encodeURIComponent(options.password || '');
+	}
+
+	const { acceptCompression = true } = options;
+	if (acceptCompression) {
+		requestOptions.headers[ 'Accept-Encoding' ] = 'gzip, deflate';
+	}
+
+	const request = parsedUrl.protocol === 'https:' ? https.request(requestOptions) : http.request(requestOptions);
+
+	const task = new Task<Response>((resolve, reject) => {
+		let timeoutHandle: Handle;
+		let timeoutReject: Function = reject;
+
+		if (options.socketOptions) {
+			if (options.socketOptions.timeout) {
+				request.setTimeout(options.socketOptions.timeout);
+			}
+
+			if ('noDelay' in options.socketOptions) {
+				request.setNoDelay(options.socketOptions.noDelay);
+			}
+
+			if ('keepAlive' in options.socketOptions) {
+				const initialDelay: number | undefined = options.socketOptions.keepAlive;
+				request.setSocketKeepAlive(initialDelay >= 0, initialDelay);
+			}
+		}
+
+		request.once('response', (message: http.IncomingMessage) => {
+			const response = new NodeResponse(message);
+
+			// Redirection handling defaults to true in order to harmonise with the XHR provider, which will always
+			// follow redirects
+			if (
+				response.status >= 300 &&
+				response.status < 400
+			) {
+				const redirectOptions = options.redirectOptions || {};
+				const newOptions = Object.create(options);
```

Comment:
`deepAssign({}, options)` would also work. I figured there was a reason the code was using `Object.create()` instead of `Object.assign()`, so to keep the code somewhat the same, I was noting that we don't have the `Object.create()` counterpart in `@dojo/lang`. Once you start iterating over properties, you may as well use `deepAssign()` anyway.